File: app/src/main/java/org/fossasia/susi/ai/rest/model/SusiResponse.java
package org.fossasia.susi.ai.rest.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class SusiResponse {
    @SerializedName("client_id")
    @Expose
    String clientId;
    @SerializedName("query")
    @Expose
    String query;
    @SerializedName("query_date")
    @Expose
    String queryDate;
    @SerializedName("answer_date")
    @Expose
    String answerDate;
    @SerializedName("answer_time")
    @Expose
    Integer answerTime;
    @SerializedName("count")
    @Expose
    Integer count;
    @SerializedName("answers")
    @Expose
    List<Answer> answers = new ArrayList<Answer>();
    @SerializedName("session")
    @Expose
    Session session;

    public String getClientId() {
        return clientId;
    }

    public String getQuery() {
        return query;
    }

    public String getQueryDate() {
        return queryDate;
    }

    public String getAnswerDate() {
        return answerDate;
    }

    public Integer getAnswerTime() {
        return answerTime;
    }

    public Integer getCount() {
        return count;
    }

    public List<Answer> getAnswers() {
        return answers;
    }

    public Session getSession() {
        return session;
    }
}

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/SusiResponse.kt
package org.fossasia.susi.ai.rest.responses.susi

import com.google.gson.annotations.SerializedName
import java.util.ArrayList

data class SusiResponse(
    @SerializedName("client_id")
    val clientId: String? = null,
    val query: String = "",
    @SerializedName("query_date")
    val queryDate: String = "",
    @SerializedName("answer_date")
    val answerDate: String = "",
    @SerializedName("answer_time")
    val answerTime: Int = 0,
    val count: Int = 0,
    val answers: List<Answer> = ArrayList(),
    val session: Session? = null
)
